Reorder the Fx execution order to in-time get_attr rather than putting all get_attr ahead

RenfeiChen-FB opened the pull request:
Summary:
Basically today we:

[getattr....getattr, call partition1, call parition2]

this makes getattr just in time:

so [getattr, call partition1, getattr, call partition 2 ..]

Test Plan:
CMF and MAI test result:
https://fb.quip.com/K5J9A7G246Ox

Reviewed By: xush6528

Differential Revision: D43074214

Skylion007 commented on torch/fx/passes/split_module.py:
```diff
@@ -136,6 +136,31 @@ def forward(self, x, y):
+            attr_val = m
+            for atom in node.target.split("."):
+                if not hasattr(attr_val, atom):
+                    raise RuntimeError(f"Node target {node.target} not found!")
```
Shouldn't this be an attribute AttributeError?

RenfeiChen-FB committed:
Reorder the Fx execution order to in-time get_attr rather than putting all get_attr ahead (#94996)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/94996

Basically today we:

[getattr....getattr, call partition1, call parition2]

this makes getattr just in time:

so [getattr, call partition1, getattr, call partition 2 ..]

Test Plan:
CMF and MAI test result:
https://fb.quip.com/K5J9A7G246Ox

Reviewed By: xush6528, angelayi

Differential Revision: D43074214

fbshipit-source-id: e2665c6d98c1bb242a4268450da6ec4895fbb7dd
